runtime/src/serde_snapshot.rs

```diff
@@ -146,18 +200,33 @@ where
 {
     macro_rules! INTO {
         ($x:ident) => {{
-            let (bank_fields, accounts_db_fields) = $x::deserialize_bank_fields(stream)?;
-
+            let (full_snapshot_bank_fields, full_snapshot_accounts_db_fields) =
+                $x::deserialize_bank_fields(snapshot_streams.full_snapshot_stream)?;
+            let (incremental_snapshot_bank_fields, incremental_snapshot_accounts_db_fields) =
+                if let Some(ref mut incremental_snapshot_stream) =
+                    snapshot_streams.incremental_snapshot_stream
+                {
+                    let (bank_fields, accounts_db_fields) =
+                        $x::deserialize_bank_fields(incremental_snapshot_stream)?;
+                    (Some(bank_fields), Some(accounts_db_fields))
+                } else {
+                    (None, None)
+                };
```

Comment:
I use this pattern a number of times to handle `Option<>` on incremental snapshot stuffs. While this way works, it feels very redundant and verbose. Is there a better/clearer way to do this?

Reply:
This seems to make sense to me, the `?` in `$x::deserialize_bank_fields(incremental_snapshot_stream)?;` makes it hard to use combinators

Reply:
Yeah... I was hoping there was some other Rust magic out there :)